packages/dialog/src/mwc-dialog-base.ts

```diff
@@ -0,0 +1,301 @@
+      areButtonsStacked: () => this.stacked,
+      clickDefaultButton: () => {
+        const primary = this.primaryButton;
+        if (primary) {
+          primary.click();
+        }
+      },
+      eventTargetMatches: (target, selector) =>
+          target ? (target as Element).matches(selector) : false,
+      getActionFromEvent: (e: Event) => {
+        if (!e.target) {
+          return '';
+        }
+
+        const element =
+            (e.target as Element).closest(`[${this.actionAttribute}]`);
+        const action = element && element.getAttribute(this.actionAttribute);
+        return action;
+      },
+      getInitialFocusEl: () => {
+        return this.getInitialFocusEl();
+      },
+      isContentScrollable: () => {
+        const el = this.contentElement;
+        return el ? el.scrollHeight > el.offsetHeight : false;
+      },
+      notifyClosed: (action) => this.emitNotification('closed', action),
+      notifyClosing: (action) => this.emitNotification('closing', action),
+      notifyOpened: () => this.emitNotification('opened'),
+      notifyOpening: () => this.emitNotification('opening'),
+      reverseButtons: () => this.actionsReversed = !this.actionsReversed,
+      releaseFocus: () => {
+        blockingElements.remove(this);
+      },
+      trapFocus: (el) => {
+        blockingElements.push(this);
+        if (el) {
+          el.focus();
+        }
+      },
+    };
+  }
+
+  protected render() {
+    const classes = {
+      [cssClasses.STACKED]: this.stacked,
+    };
+
+    let title = html``;
+
+    if (this.title) {
+      title = html`<h2 class="mdc-dialog__title">${this.title}</h2>`;
+    }
+
+    const actoinsClasses = {
+      'mdc-dialog__actions': !this.hideActions,
+    };
+
+    let slotNames = ['secondaryAction', 'primaryAction'];
+
+    if (this.actionsReversed) {
+      slotNames = slotNames.reverse();
+    }
+
+    const actionSlots = html`
+        <slot name="${slotNames[0]}"></slot>
+        <slot name="${slotNames[1]}"></slot>`;
+
+    return html`
+    <div class="mdc-dialog ${classMap(classes)}"
+        role="alertdialog"
+        aria-modal="true"
+        aria-labelledby="my-dialog-title"
+        aria-describedby="my-dialog-content"
+        mdc>
+      <div class="mdc-dialog__container">
+        <div class="mdc-dialog__surface">
+          ${title}
+          <div class="mdc-dialog__content">
+            <slot></slot>
+          </div>
+          <footer
+              id="actions"
+              class="${classMap(actoinsClasses)}">
+            ${actionSlots}
+          </footer>
+        </div>
+      </div>
+      <div class="mdc-dialog__scrim"></div>
+    </div>`;
+  }
+
+  firstUpdated() {
+    super.firstUpdated();
+    this.mdcFoundation.setAutoStackButtons(true);
+
+    this.originalOpen = this.mdcFoundation.open.bind(this.mdcFoundation);
+    this.originalClose = this.mdcFoundation.close.bind(this.mdcFoundation);
+
+    this.mdcFoundation.open = () => {
```

Comment:
This doesn't feel good. Is there no way to know when the foundation has opened the dialog?